docs/UserGuide.md

```diff
@@ -197,6 +205,27 @@ Exits the program.
 Format: `exit`
 
 
+### Undo last command : `undo`
+
+Undoes the previous command by reverting the current data displayed to the state it was in before the last command was executed.
+
+Format: `undo`
+
+* If there is a previous state available, the current state is reverted to that state
+* If the current state is the earliest possible one, it shows a message informing the user that there is nothing more to undo
+
+
+### Redo last command : `redo`
+
+Redoes the last undone command by reverting the current data displayed to the state it was in before the last undo command was executed.
+
+Format: `redo`
+
+* If there is an undone state available, the current state is reverted to that state
+* If the current state is the latest possible one, it shows a message informing the user that there is nothing more to redo
+* After any command that changes the state of data (such as add, clear, delete, edit), the new state becomes the latest state
+(i.e. the previous undo commands are "forgotten" and `redo` will have no effect)
+
 
 ### Saving the data
```

Comment:
Do you want to include the key arrow up feature to go back to previous command entered in commandBox? Trivial but like cool only

Reply:
I believe it already has been added by Xing Yu in the next paragraph